src/encoded/types/file.py

```diff
@@ -919,8 +922,32 @@ def build_key(cls, registry, uuid, properties):
 
     @classmethod
     def build_external_creds(cls, registry, uuid, properties):
```

Comment:
```suggestion
    def build_external_creds(cls, registry, uuid, properties, try_buckets=None):
        if try_buckets is None:
            try_buckets = [registry.settings['file_wfout_bucket'], registry.settings['file_upload_bucket']]
```
so later you can iterate over those. I think it will make it easier to test and possibly more reusable later.